client/main.lua

```diff
@@ -1,100 +1,136 @@
+                setMultipliers(cache.vehicle, false)
                 SetEntityMaxSpeed(cache.vehicle, 999.0)
-                Entity(cache.vehicle).state:set("nitro", Entity(cache.vehicle).state.nitro - 0.25, true)
+                StartScreenEffect("RaceTurbo", 0, 0)
+                vehicleState:set("nitro", vehicleState.nitro - 0.25, true)
             else
-                SetVehicleBoostActive(cache.vehicle, 0)
-                SetVehicleEnginePowerMultiplier(cache.vehicle, 1.0)
-                SetVehicleEngineTorqueMultiplier(cache.vehicle, 1.0)
-                for index, _ in pairs(Fxs) do
-                    StopParticleFxLooped(Fxs[index], 1)
-                    TriggerServerEvent('qbx_nitro:server:StopSync', Plate)
-                    Fxs[index] = nil
-                end
-                StopScreenEffect("RaceTurbo")
-                TriggerServerEvent('qbx_nitro:server:UnloadNitrous', Plate)
-                nitrousActivated = false
+                stopBoosting(cache.vehicle)
+                vehicleState:set("nitro", 0, true)
             end
             if IsControlJustReleased(0, 36) and cache.seat == -1 then
-                SetVehicleBoostActive(cache.vehicle, 0)
-                SetVehicleEnginePowerMultiplier(cache.vehicle, 1.0)
-                SetVehicleEngineTorqueMultiplier(cache.vehicle, 1.0)
-                for index, _ in pairs(Fxs) do
-                    StopParticleFxLooped(Fxs[index], 1)
-                    TriggerServerEvent('qbx_nitro:server:StopSync', Plate)
-                    Fxs[index] = nil
-                end
-                StopScreenEffect("RaceTurbo")
-                nitrousActivated = false
+                stopBoosting(cache.vehicle)
             end
             Wait(0)
         end
     end)
 end
 
+AddStateBagChangeHandler('nitroFlames', nil, function(bagName, _, value)
+    local veh = GetEntityFromStateBagName(bagName)
+    if value then
+        syncFlames(veh)
+    else
+        local plate = trim(GetVehicleNumberPlateText(veh))
+        stopSync(plate)
+    end
+end)
+
+local NitrousLoop = false
 local function nitrousLoop()
+    if not cache.vehicle or cache.seat ~= -1 then return end
     local sleep = 0
+    NitrousLoop = true
     CreateThread(function()
-        while cache.vehicle do
-            if (Entity(cache.vehicle)?.state?.nitro or 0) > 0 then
+        while cache.vehicle and NitrousLoop do
+            if IsVehicleEngineOn(cache.vehicle) and (Entity(cache.vehicle)?.state?.nitro or 0) > 0 then
                 sleep = 0
-                if IsControlJustPressed(0, 36) and cache.seat == -1 and not nitroDelay then
-                    TriggerServerEvent('qbx_nitro:server:SyncFlames', VehToNet(cache.vehicle))
-                    nitrousActivated = true
-                    loadEffects()
+                if IsControlJustPressed(0, 36) and not nitroDelay then
+                    Entity(cache.vehicle).state:set("nitroFlames", true, true)

```

Comment:
Localize `Entity(cache.vehicle)` so we don't intiailize the state every frame (very expensive)